app/models/manageiq/providers/vmware/infra_manager/refresh_parser.rb

```diff
@@ -1272,6 +1273,25 @@ def self.link_root_folder(data)
         end
       end
 
+      def self.set_hidden_folders(data)
+        return if data[:ems_root].nil?
+
+        # Mark the root folder as hidden
+        data[:ems_root][:hidden] = true
+
+        # Mark all child folders of each Datacenter as hidden
+        # e.g.: "vm", "host", "datastore"
+        data[:folders].select { |f| f[:type] == "Datacenter" }.each do |dc|
+          dc_children = dc.fetch_path(:ems_children, :folders)
+          dc_children.to_miq_a.each do |f|
+            f[:hidden] = true
+          end
+        end
+
+        # Set all other folders to be not hidden
+        data[:folders].each { |f| f[:hidden] = false unless f[:hidden] }
```

Comment:
`set_default_rps` uses a similar pattern to the above (doing a final loop to set all to `false`).  I'll change that in a separate PR.